### ISSUE TITLE
r2dbc-mariadb throws IllegalReferenceCountException

### ISSUE THREAD

**Issue by nakamura-to** [komapper/komapper#428]
### Describe the bug

r2dbc-mariadb throws `io.netty.util.IllegalReferenceCountException` in most test cases.

### Expected behavior

It works without exception.

### Actual behavior

<details>
<summary>stacktrace</summary>

```
io.netty.util.IllegalReferenceCountException: refCnt: 0
	at app//io.netty.buffer.AbstractByteBuf.ensureAccessible(AbstractByteBuf.java:1454)
	at app//io.netty.buffer.AbstractByteBuf.checkReadableBytes0(AbstractByteBuf.java:1440)
	at app//io.netty.buffer.AbstractByteBuf.readByte(AbstractByteBuf.java:730)
	at app//io.netty.buffer.AbstractByteBuf.readUnsignedByte(AbstractByteBuf.java:744)
	at app//org.mariadb.r2dbc.codec.TextRowDecoder.setPosition(TextRowDecoder.java:78)
	at app//org.mariadb.r2dbc.codec.TextRowDecoder.get(TextRowDecoder.java:19)
	at app//org.mariadb.r2dbc.MariadbReadable.get(MariadbReadable.java:33)
	at app//io.r2dbc.spi.Readable.get(Readable.java:53)
	at app//org.komapper.r2dbc.R2dbcAbstractType.getValue(R2dbcDataType.kt:83)
	at app//org.komapper.r2dbc.R2dbcAbstractDialect.getValue(R2dbcDialect.kt:96)
	at app//org.komapper.r2dbc.dsl.runner.R2dbcPropertyMapper.execute(R2dbcPropertyMapper.kt:12)
	at app//org.komapper.r2dbc.dsl.runner.R2dbcEntityMapper.execute(R2dbcEntityMapper.kt:14)
	at app//org.komapper.r2dbc.dsl.runner.R2dbcRowTransformers$singleEntity$1.invoke(R2dbcRowTransformers.kt:15)
	at app//org.komapper.r2dbc.dsl.runner.R2dbcRowTransformers$singleEntity$1.invoke(R2dbcRowTransformers.kt:13)
	at app//org.komapper.r2dbc.R2dbcExecutor$executeQuery$1$1$1.invokeSuspend$lambda-0(R2dbcExecutor.kt:44)
	at app//org.mariadb.r2dbc.MariadbResult.lambda$map$1(MariadbResult.java:121)
	at app//reactor.core.publisher.FluxMap$MapSubscriber.onNext(FluxMap.java:106)
	at app//reactor.core.publisher.FluxMap$MapSubscriber.onNext(FluxMap.java:120)
	at app//reactor.core.publisher.FluxFilter$FilterSubscriber.onNext(FluxFilter.java:113)
	at app//reactor.core.publisher.FluxHandle$HandleConditionalSubscriber.onNext(FluxHandle.java:327)
	at app//reactor.core.publisher.FluxTakeUntil$TakeUntilPredicateSubscriber.onNext(FluxTakeUntil.java:84)
	at app//reactor.core.publisher.FluxWindowPredicate$WindowFlux.drainRegular(FluxWindowPredicate.java:667)
	at app//reactor.core.publisher.FluxWindowPredicate$WindowFlux.drain(FluxWindowPredicate.java:745)
	at app//reactor.core.publisher.FluxWindowPredicate$WindowFlux.subscribe(FluxWindowPredicate.java:820)
	at app//reactor.core.publisher.Flux.subscribe(Flux.java:8402)
	at app//kotlinx.coroutines.reactive.PublisherAsFlow.collectImpl(ReactiveFlow.kt:94)
	at app//kotlinx.coroutines.reactive.PublisherAsFlow.collect(ReactiveFlow.kt:79)
	at app//kotlinx.coroutines.flow.FlowKt__CollectKt.emitAll(Collect.kt:132)
	at app//kotlinx.coroutines.flow.FlowKt.emitAll(Unknown Source)
	at app//kotlinx.coroutines.flow.FlowKt__MergeKt$flattenConcat$lambda-1$$inlined$collect$1.emit(Collect.kt:135)
	at app//kotlinx.coroutines.flow.FlowKt__MergeKt$flatMapConcat$$inlined$map$1$2.emit(Collect.kt:136)
	at app//kotlinx.coroutines.reactive.PublisherAsFlow.collectImpl(ReactiveFlow.kt:100)
	at app//kotlinx.coroutines.reactive.PublisherAsFlow.access$collectImpl(ReactiveFlow.kt:48)
	at app//kotlinx.coroutines.reactive.PublisherAsFlow$collectImpl$1.invokeSuspend(ReactiveFlow.kt)
	at ???(Coroutine boundary.?(?)
	at org.komapper.tx.r2dbc.TransactionManagerImpl.begin(R2dbcTransactionManager.kt:102)
	at integration.r2dbc.R2dbcUtilityKt$inTransaction$1.invokeSuspend(R2dbcUtility.kt:12)
	at integration.r2dbc.R2dbcUtilityKt$runBlockingWithTimeout$1.invokeSuspend(R2dbcUtility.kt:21)
Caused by: io.netty.util.IllegalReferenceCountException: refCnt: 0
	at app//io.netty.buffer.AbstractByteBuf.ensureAccessible(AbstractByteBuf.java:1454)
	at app//io.netty.buffer.AbstractByteBuf.checkReadableBytes0(AbstractByteBuf.java:1440)
	at app//io.netty.buffer.AbstractByteBuf.readByte(AbstractByteBuf.java:730)
	at app//io.netty.buffer.AbstractByteBuf.readUnsignedByte(AbstractByteBuf.java:744)
	at app//org.mariadb.r2dbc.codec.TextRowDecoder.setPosition(TextRowDecoder.java:78)
	at app//org.mariadb.r2dbc.codec.TextRowDecoder.get(TextRowDecoder.java:19)
	at app//org.mariadb.r2dbc.MariadbReadable.get(MariadbReadable.java:33)
	at app//io.r2dbc.spi.Readable.get(Readable.java:53)
	at app//org.komapper.r2dbc.R2dbcAbstractType.getValue(R2dbcDataType.kt:83)
	at app//org.komapper.r2dbc.R2dbcAbstractDialect.getValue(R2dbcDialect.kt:96)
	at app//org.komapper.r2dbc.dsl.runner.R2dbcPropertyMapper.execute(R2dbcPropertyMapper.kt:12)
	at app//org.komapper.r2dbc.dsl.runner.R2dbcEntityMapper.execute(R2dbcEntityMapper.kt:14)
	at app//org.komapper.r2dbc.dsl.runner.R2dbcRowTransformers$singleEntity$1.invoke(R2dbcRowTransformers.kt:15)
	at app//org.komapper.r2dbc.dsl.runner.R2dbcRowTransformers$singleEntity$1.invoke(R2dbcRowTransformers.kt:13)
	at app//org.komapper.r2dbc.R2dbcExecutor$executeQuery$1$1$1.invokeSuspend$lambda-0(R2dbcExecutor.kt:44)
	at app//org.mariadb.r2dbc.MariadbResult.lambda$map$1(MariadbResult.java:121)
	at app//reactor.core.publisher.FluxMap$MapSubscriber.onNext(FluxMap.java:106)
	at app//reactor.core.publisher.FluxMap$MapSubscriber.onNext(FluxMap.java:120)
	at app//reactor.core.publisher.FluxFilter$FilterSubscriber.onNext(FluxFilter.java:113)
	at app//reactor.core.publisher.FluxHandle$HandleConditionalSubscriber.onNext(FluxHandle.java:327)
	at app//reactor.core.publisher.FluxTakeUntil$TakeUntilPredicateSubscriber.onNext(FluxTakeUntil.java:84)
	at app//reactor.core.publisher.FluxWindowPredicate$WindowFlux.drainRegular(FluxWindowPredicate.java:667)
	at app//reactor.core.publisher.FluxWindowPredicate$WindowFlux.drain(FluxWindowPredicate.java:745)
	at app//reactor.core.publisher.FluxWindowPredicate$WindowFlux.subscribe(FluxWindowPredicate.java:820)
	at app//reactor.core.publisher.Flux.subscribe(Flux.java:8402)
	at app//kotlinx.coroutines.reactive.PublisherAsFlow.collectImpl(ReactiveFlow.kt:94)
	at app//kotlinx.coroutines.reactive.PublisherAsFlow.collect(ReactiveFlow.kt:79)
	at app//kotlinx.coroutines.flow.FlowKt__CollectKt.emitAll(Collect.kt:132)
	at app//kotlinx.coroutines.flow.FlowKt.emitAll(Unknown Source)
	at app//kotlinx.coroutines.flow.FlowKt__MergeKt$flattenConcat$lambda-1$$inlined$collect$1.emit(Collect.kt:135)
	at app//kotlinx.coroutines.flow.FlowKt__MergeKt$flatMapConcat$$inlined$map$1$2.emit(Collect.kt:136)
	at app//kotlinx.coroutines.reactive.PublisherAsFlow.collectImpl(ReactiveFlow.kt:100)
	at app//kotlinx.coroutines.reactive.PublisherAsFlow.access$collectImpl(ReactiveFlow.kt:48)
	at app//kotlinx.coroutines.reactive.PublisherAsFlow$collectImpl$1.invokeSuspend(ReactiveFlow.kt)
	at app//kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at app//kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at app//kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:274)
	at app//kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85)
	at app//kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
	at app//kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
	at app//kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)
	at app//kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
	at app//integration.r2dbc.R2dbcUtilityKt.runBlockingWithTimeout(R2dbcUtility.kt:20)
	at app//integration.r2dbc.R2dbcUtilityKt.runBlockingWithTimeout$default(R2dbcUtility.kt:19)
	at app//integration.r2dbc.R2dbcUtilityKt.inTransaction(R2dbcUtility.kt:11)
	at app//integration.r2dbc.SelectTest.list(SelectTest.kt:30)
	at java.base@11.0.6/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base@11.0.6/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base@11.0.6/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base@11.0.6/java.lang.reflect.Method.invoke(Method.java:566)
	at app//org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:688)
	at app//org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at app//org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at app//org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:149)
	at app//org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:140)
	at app//org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:84)
	at app//org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
	at app//org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at app//org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at app//org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at app//org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at app//org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at app//org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:104)
	at app//org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:98)
	at app//org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$6(TestMethodTestDescriptor.java:210)
	at app//org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at app//org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:206)
	at app//org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:131)
	at app//org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:65)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
	at app//org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
	at app//org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
	at app//org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
	at java.base@11.0.6/java.util.ArrayList.forEach(ArrayList.java:1540)
	at app//org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:143)
	at app//org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
	at app//org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
	at app//org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
	at java.base@11.0.6/java.util.ArrayList.forEach(ArrayList.java:1540)
	at app//org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:143)
	at app//org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:129)
	at app//org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:127)
	at app//org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:126)
	at app//org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:84)
	at app//org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
	at app//org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at app//org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:108)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:96)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:75)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.processAllTestClasses(JUnitPlatformTestClassProcessor.java:99)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.access$000(JUnitPlatformTestClassProcessor.java:79)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor.stop(JUnitPlatformTestClassProcessor.java:75)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.stop(SuiteTestClassProcessor.java:61)
	at java.base@11.0.6/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base@11.0.6/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base@11.0.6/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base@11.0.6/java.lang.reflect.Method.invoke(Method.java:566)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at com.sun.proxy.$Proxy2.stop(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker$3.run(TestWorker.java:193)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.executeAndMaintainThreadName(TestWorker.java:129)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:100)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:60)
	at org.gradle.process.internal.worker.child.ActionExecutionWorker.execute(ActionExecutionWorker.java:56)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:133)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:71)
	at app//worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:69)
	at app//worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:74)
```
</details>

### Steps to reproduce the problem

Execute following commands:

```sh
$ git clone https://github.com/komapper/komapper.git
$ cd komapper
$ ./gradlew :integration-test-r2dbc:mariadb
```

### Versions

- Komapper: 0.28.0
- Kotlin: 1.5.31
- JVM: temurin 11
- JDBC Driver or R2DBC Driver: r2dbc-mariadb 1.1.1-rc
- Database: MariaDB 10.6.3
- OS: Mac OS X 11.1 x86_64

**Comment by nakamura-to** [comment]
Related information:

- https://jira.mariadb.org/browse/R2DBC-51
- mariadb-corporation/mariadb-connector-r2dbc#24

**Comment by nakamura-to** [comment]
This issue has been resolved by updating to r2dbc-mariadb v1.1.3 #830
